lib/Dialect/FIRRTL/Transforms/Dedup.cpp

```diff
@@ -442,9 +402,8 @@ struct Deduper {
       instAnnos.applyToOperation(inst);
     }
     // Erase the NLA from the leaf module's nlaMap.
-    llvm::erase_value(nlaMap[nla.leafMod()], nla);
     targetMap.erase(nla.getNameAttr());
-    nla->erase();
+    nlaTable->erase(nla);
```

Comment:
there's a symbol table floating around, it needs to no about this too (and it can erase things for you I think).